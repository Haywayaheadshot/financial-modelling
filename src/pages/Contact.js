import React from 'react';
import '../styles/contact.css';

export default function Contact() {
  return (
    <div>
      <h1 className="contact-header">Thank you for checking out my project</h1>
      <p className="contact-thanks-note">
        I will like to thank Microverse for impacting this knowledge.
        Thank you to my colleagues Diego Garcia and Jose Zepada who
        helped me when I had blockers. If you
        loved this project please do not hesitate to give a STAR on Github.
      </p>
      <div className="contact-authors">
        <h1>Authors Github Usernames</h1>
        <br />
        <h3>ABUBAKAR UMMAR: @Haywayaheadshot</h3>
        <h3>DIEGO GARCIA: @taldr27</h3>
        <h3>JOSE ZEPADA: @J2ZROMERO</h3>
      </div>
    </div>
  );
}
