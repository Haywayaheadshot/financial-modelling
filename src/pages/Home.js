import React from 'react';
import HomeThumbnail from '../components/HomeThumbnail';

export default function Home() {
  return (
    <div>
      <HomeThumbnail />
    </div>
  );
}
