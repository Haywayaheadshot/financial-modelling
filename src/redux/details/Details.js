const ADD_DETAILS = 'FINANCIAL-MODELLING/src/redux/details/addDetails';

export const addDetails = () => (
    {
      type: ADD_DETAILS,
      payload: test,
    }
);

export default function booksReducer(state = [], action) {
    switch (action.type) {
      case ADD_DETAILS:
        return state.filter((coin) => coin.id === action.payload);
      default:
        return state;
    }
  }
