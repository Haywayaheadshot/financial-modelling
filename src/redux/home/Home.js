import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const GET_COINS = 'FINANCIAL-MODELLING/src/redux/home/getCoins';

const initialState = [];

const getCoinsApi = 'https://api.coinstats.app/public/v1/coins?skip=0&limit=5&currency=EUR';

// action creators for display, join and leave missions
export const getCoins = createAsyncThunk(
  GET_COINS, () => axios.get(getCoinsApi).then((res) => {
    const { coins } = res.data;
    return coins;
  }),
);

const coinsSlice = createSlice({
  name: 'coins',
  initialState,
  extraReducers: (builder) => {
    builder.addCase(getCoins.fulfilled, (_, action) => action.payload);
    builder.addCase(getCoins.rejected, (state) => {
      const newState = state;
      newState.status = 'failed';
    });
    builder.addCase(getCoins.pending, (_, action) => action.payload);
  },
});

export default coinsSlice.reducer;
