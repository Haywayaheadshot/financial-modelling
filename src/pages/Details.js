import React, { useEffect } from 'react';
import { NavLink, useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { getCoins } from '../redux/home/Home';

export default function Details() {
  const params = useParams();
  const coins = useSelector((state) => state.coins);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getCoins());
  }, []);
  const filteredCoin = coins.filter((coin) => (coin.id === params.id));
  return (
    <div>
      <NavLink to="../">
        <button type="button">Back</button>
      </NavLink>
      { filteredCoin.map((coin) => (
        <section key={coin.id} id={coin.id}>
          <div>
            <h1>
              {coin.name}
              /BTC
            </h1>
            <h2>Rank</h2>
          </div>
          <table>
            <tbody>
              <tr>
                <td className="column-data">{`Total Supply: ${coin.totalSupply}`}</td>
                <td className="column-data">102220</td>
              </tr>
              <tr>
                <td className="column-data">Price Against BTC</td>
                <td className="column-data">102220</td>
              </tr>
              <tr>
                <td className="column-data">Volume</td>
                <td className="column-data">102220</td>
              </tr>
              <tr>
                <td className="column-data">Market Cap</td>
                <td className="column-data">102220</td>
              </tr>
              <tr>
                <td className="column-data">Available Supply</td>
                <td className="column-data">102220</td>
              </tr>
              <tr>
                <td className="column-data">Total Supply</td>
                <td className="column-data">102220</td>
              </tr>
              <tr>
                <td className="column-data">Hourly Price Change</td>
                <td className="column-data">102220</td>
              </tr>
              <tr>
                <td className="column-data">Weekly Price Change</td>
                <td className="column-data">102220</td>
              </tr>
              <tr>
                <td className="column-data">Daily Price Change</td>
                <td className="column-data">102220</td>
              </tr>
            </tbody>
          </table>
        </section>
      ))}
    </div>
  );
}
