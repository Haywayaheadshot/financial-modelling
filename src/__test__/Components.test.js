import { render, screen } from '@testing-library/react';
import { StrictMode } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Home from '../pages/Home';
import Contact from '../pages/Contact';
import Details from '../pages/Details';
import '@testing-library/jest-dom';

describe('Testing Home page', () => {
  it('Should match the snapshot', () => {
    const home = render(
      <StrictMode>
        <Router>
          <Provider store={store}>
            <Home />
          </Provider>
        </Router>
      </StrictMode>,
    );
    expect(home).toMatchSnapshot();
  });

  // it('Should have text Bitcoin', () => {
  //   render(
  //     <Provider store={store}>
  //       <Router>
  //         <Home />
  //       </Router>
  //     </Provider>,
  //   );
  //   expect(screen.getByText('Bitcoin')).toBeInTheDocument();
  // });

  it('Should have text Ethereum', () => {
    render(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    );
    expect(screen.getByText('Ethereum')).toBeInTheDocument();
  });
});

describe('Testing Contacts page', () => {
  it('Should match the companies snap', () => {
    const contact = render(
      <StrictMode>
        <Router>
          <Provider store={store}>
            <Contact />
          </Provider>
        </Router>
      </StrictMode>,
    );
    expect(contact).toMatchSnapshot();
  });
});

describe('Testing Details page', () => {
  it('Should match the details snap', () => {
    const details = render(
      <StrictMode>
        <Router>
          <Provider store={store}>
            <Details />
          </Provider>
        </Router>
      </StrictMode>,
    );
    expect(details).toMatchSnapshot();
  });
});
