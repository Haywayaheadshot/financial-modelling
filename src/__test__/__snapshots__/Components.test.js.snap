// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Contacts page Should match the companies snap 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1
          class="contact-header"
        >
          Thank you for checking out my project
        </h1>
        <p
          class="contact-thanks-note"
        >
          I will like to thank Microverse for impacting this knowledge. Thank you to my colleagues Diego Garcia and Jose Zepada who helped me when I had blockers. If you loved this project please do not hesitate to give a STAR on Github.
        </p>
        <div
          class="contact-authors"
        >
          <h1>
            Authors Github Usernames
          </h1>
          <br />
          <h3>
            ABUBAKAR UMMAR: @Haywayaheadshot
          </h3>
          <h3>
            DIEGO GARCIA: @taldr27
          </h3>
          <h3>
            JOSE ZEPADA: @J2ZROMERO
          </h3>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1
        class="contact-header"
      >
        Thank you for checking out my project
      </h1>
      <p
        class="contact-thanks-note"
      >
        I will like to thank Microverse for impacting this knowledge. Thank you to my colleagues Diego Garcia and Jose Zepada who helped me when I had blockers. If you loved this project please do not hesitate to give a STAR on Github.
      </p>
      <div
        class="contact-authors"
      >
        <h1>
          Authors Github Usernames
        </h1>
        <br />
        <h3>
          ABUBAKAR UMMAR: @Haywayaheadshot
        </h3>
        <h3>
          DIEGO GARCIA: @taldr27
        </h3>
        <h3>
          JOSE ZEPADA: @J2ZROMERO
        </h3>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Details page Should match the details snap 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="details-container"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <button
            class="back-button"
            type="button"
          >
            Back
          </button>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="details-container"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <button
          class="back-button"
          type="button"
        >
          Back
        </button>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Home page Should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <div>
            <input
              class="search-input"
              placeholder="Find a title ..."
            />
          </div>
          <div
            class="home-coin-container"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <div>
          <input
            class="search-input"
            placeholder="Find a title ..."
          />
        </div>
        <div
          class="home-coin-container"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
